runtime/lib/capi/ExecutionContext.hpp

```diff
@@ -127,7 +185,14 @@ class ExecutionContext final {
 
     [[nodiscard]] auto getDeviceRecorderStatus() const -> bool { return _tape_recording; }
 
-    [[nodiscard]] bool initDevice(std::string_view name) noexcept
```

Comment:
One feature that's missing is the ability of using one shared library for multiple devices. Let's say if we want to build multiple Lightning simulators in one shared library, users should be still able to provide the name of the backend device as well as the path to the shared library to properly load/unload the QuantumDevice implementation.

I'm not sure how important this feature would be in practice but it will have one direct use case for Lightning simulators.

Reply:
Agree! Moving towards `initVendorCliend` is on the roadmap. But, I'd like to know if you have already some ideas that how the support may look like if we want to get this support (i.e., having one shared library for multiple devices) in?

Reply:
I think it doesn't matter whether it is one shared library with multiple devices or one shared library per device. What matters is one library per vendor and each vendor is free to implement the client however they want. (Single shared library vs multiple shared libraries). I think we need to think a bit more about what the vendor client would look like to have a good design. I have more experience with OpenCL so I am a bit biased towards thinking about their API. However, I am interested to learn about alternatives as well.

For the time being, the API as implemented here indeed follows the one shared library per device assumption. Which I do not think is necessarily bad as a first version.

Do you have plans for keeping `librt_backend` together?

Reply:
I don't have any points against this design at the moment! This is actually what we've been discussing it for a while :) With this suggestion, I was trying to rethink about the one-to-many support and its use cases. Not a blocker for your work here. being said, this satisfies the requirements and is aligned with the roadmap. :tada:

Reply:
No, I don't plan to keep `librt_backend` for the scope of the work.